vicinity/backends/voyager.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Union
+
+import numpy as np
+from numpy import typing as npt
+from voyager import Index, Space
+
+from vicinity.backends.base import AbstractBackend, BaseArgs
+from vicinity.datatypes import Backend, QueryResult
+from vicinity.utils import Metric, normalize
+
+
+@dataclass
+class VoyagerArgs(BaseArgs):
+    dim: int = 0
+    metric: str = "cosine"
+    ef_construction: int = 200
+    m: int = 16
+
+
+class VoyagerBackend(AbstractBackend[VoyagerArgs]):
+    argument_class = VoyagerArgs
+    supported_metrics = {Metric.COSINE, Metric.EUCLIDEAN}
+    inverse_metric_mapping = {
+        Metric.COSINE: "cosine",
+        Metric.EUCLIDEAN: "l2",
+    }
+
+    metric_int_mapping = {
+        "l2": 0,
+        "cosine": 2,
+    }
```

Comment:
I think inner_product is also supported in Voyager, I would also add that here as one of the metrics as we also support it (see https://github.com/MinishLab/vicinity/blob/main/vicinity/utils.py#L63).

Reply:
Actually I'm not 100% sure if that is actually a distance metric. Can ignore this for this PR.

Reply:
okay